src/nunavut/lang/cpp/support/serialization.j2

```diff
@@ -154,7 +154,16 @@ public:
         else { new(error_ptr()) Error(*other.error_ptr()); }
     }
     expected& operator=(const expected& other){
-        this->~expected(); return *new(this) expected(other);
+        this->~expected();
+        return *new(this) expected(other);
```

Comment:
Don't you need to call the destructor on the current object before you do the inplace new?

Reply:
This is exactly what this code is doing - destroying old object and copy-constructing the new one inplace. This placement new is running on the object itself (`expected<T>`), not `T` or `Error`.

Reply:
When you `new` on the object itself you don't have a way to ensure any resources acquired in the ctor are released before they are reacquired. If you make this class `final` then you can rely on the implementation but if you allow subclassing you have to handle this somehow.

Reply:
I've made all serialization library types final.